core/store/src/config.rs

```diff
@@ -66,6 +66,19 @@ pub struct StoreConfig {
     /// This config option is temporary and will be removed once flat storage is implemented.
     pub sweat_prefetch_senders: Vec<String>,
 
+    /// List of shard UIDs for which we should load the tries in memory.
+    /// TODO(#9511): This does not automatically survive resharding. We may need to figure out a
+    /// strategy for that.
+    pub load_mem_tries_for_shards: Vec<String>,
```

Comment:
nit: name is not super clear to indicate type of this variable, especially along with the bool variable below. Maybe shards_to_load_mem_tries_for?

Reply:
It's kind of consistent with sweat_prefetch_receivers/senders though :P

Reply:
I prefer conistency with TrieCacheConfig which stores `per_shard_max_bytes` mapping ShardUId to bytes, thus using `Vec<ShardUId>` here as well.
And, hm, maybe we should have a script to upgrade shard ids in configs as well when resharding happens. Because it is never clear what should be next shard id